src/main.py

```diff
@@ -1,181 +1,11 @@
-    #TODO: make sure Lsm() is properly initialized as we may never exit this function if lsm is using the backup function values
-    lsm = Lsm()
-
-    while True:
-        if vec_len(lsm.read_acceleration_safe()) < LAUNCH_ACCELERATION_THRESHOLD:
-            break
-
-
-def powered_flight():
-    """This is when the rocket motor is actively firing.
-
-    This should only last around three seconds.
-
-    The flight computer should stay in this state until either we detect motor
-    burnout, or a manual limit is reached (such as 5s)
-
-    During this stage there should be **no airbrakes**
-    """
-    # TODO: Implement this state
-    pass
-
-
-def coast_flight():
-    """This is when the rocket is still moving upwards to apogee, but after
-    motor burnout.
-
-    This phase will last roughly 27s.
-
-    The flight computer should stay in this state until either we detect apogee,
-    or a manual limit is reached (such as 45s).
-
-    This state will be responsible for Airbrakes!
-    """
-    # TODO: Implement this state
-    # TODO: This is where we will have to deal with threading for the airbrakes
-    #       controls.
-    pass
-
-
-def recovery():
-    """This is the final state after apogee.
-
-    This phase is of unknown length (to Mateo. Should ask around to confirm.).
-
-    The flight computer should stay in this state until recovered. This is
-    forever as far as the FC is concerned.
-    """
-    # TODO: Implement this state
-    pass
-
+from flight_computer import FlightComputer
 
 def main(self):
-    """Main loop of the program.
-
-    This is where most of the magic happens and where all states are controlled.
-    """
-
-    # This is the initialization state
-    (rocket_data, lsm, bme, adx) = initialize()
-
-    # At this point we are sitting on the rail and waiting for a detection of
-    # motor ignition.
-    standby(lsm)
-
-    # These are the in-flight stages.
-    powered_flight()
-    coast_flight()
-    recovery()
-
-    # And that's a wrap.
-    print("Thank you for flying Air Xenia. Hope you enjoyed your flight.")
-
-
-def process_cli_args():
-    """Process the supplied command line arguments.
-
-
-    Flags
-    -----
-
-    * `--test` or `-t` : Run the program is test mode.
-
-
-    Returns
-    -------
-
-    The return values is a struct of (boolean), where the first element is
-    whether or not to run in test mode.
-    """
-    test_mode = False
-
-    for (i, arg) in enumerate(sys.argv):
-        if i == 0:
-            continue
-
-        if arg == "--test" or arg == "-t":
-            test_mode = True
-        else:
-            # NOTE: We explicitely DO NOT fail here because this is a rocket.
-            #       An incorrect argument is not a reason to fail completely.
-            print("Warning: Unkown argument supplied: ", arg)
-
-    return (test_mode)
-
+    flight_computer = FlightComputer()
+    flight_computer.startup()
```

Comment:
The FlightComputer init() also runs the startup function, we don't need it twice